Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/homebrew/lib/python3.10/site-packages/nbclient/client.py", line 1265, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/homebrew/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_1/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/opt/homebrew/lib/python3.10/site-packages/nbclient/client.py", line 703, in async_execute
    await self.async_execute_cell(
  File "/opt/homebrew/lib/python3.10/site-packages/nbclient/client.py", line 1021, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/homebrew/lib/python3.10/site-packages/nbclient/client.py", line 915, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
df = pd.read_parquet('../data/SO_2014_2022.pq')

avg_work_experience = df.groupby('Gender')['YearsCodePro'].mean().reset_index()

fig = px.bar(
    avg_work_experience,
    x='Gender',
    y='YearsCodePro',
    color='Gender',
    labels={
        'Gender': 'Gender',
        'YearsCodePro': 'Average years of work experience'
    },
    color_discrete_map={
        'male': '#645cff',
        'female': '#f25fe9'
    },
    title='Average Work Experience By Gender<br><sup>male survey respondents generally have more professional work experience</sup>',
)

fig.update_traces(
    width=0.4,
    showlegend=False,
    hoverlabel={'font_color': 'white', 'bordercolor': 'white'}
)

for trace in fig.data:
    series.hovertemplate = '<b>' + trace.name + '</b><br>Salary increase: %{y}<extra></extra>'

# fig.add_annotation(x=1, y=0,
#                    xref="paper", yref="paper",
#                    align='right',
#                    showarrow=False,
#                    xanchor='right', yanchor='bottom',
#                    text='some fancy caption here')

fig.show()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[3], line 28[0m
[1;32m     21[0m fig[38;5;241m.[39mupdate_traces(
[1;32m     22[0m     width[38;5;241m=[39m[38;5;241m0.4[39m,
[1;32m     23[0m     showlegend[38;5;241m=[39m[38;5;28;01mFalse[39;00m,
[1;32m     24[0m     hoverlabel[38;5;241m=[39m{[38;5;124m'[39m[38;5;124mfont_color[39m[38;5;124m'[39m: [38;5;124m'[39m[38;5;124mwhite[39m[38;5;124m'[39m, [38;5;124m'[39m[38;5;124mbordercolor[39m[38;5;124m'[39m: [38;5;124m'[39m[38;5;124mwhite[39m[38;5;124m'[39m}
[1;32m     25[0m )
[1;32m     27[0m [38;5;28;01mfor[39;00m trace [38;5;129;01min[39;00m fig[38;5;241m.[39mdata:
[0;32m---> 28[0m     [43mseries[49m[38;5;241m.[39mhovertemplate [38;5;241m=[39m [38;5;124m'[39m[38;5;124m<b>[39m[38;5;124m'[39m [38;5;241m+[39m trace[38;5;241m.[39mname [38;5;241m+[39m [38;5;124m'[39m[38;5;124m</b><br>Salary increase: [39m[38;5;124m%[39m[38;5;132;01m{y}[39;00m[38;5;124m<extra></extra>[39m[38;5;124m'[39m
[1;32m     30[0m [38;5;66;03m# fig.add_annotation(x=1, y=0,[39;00m
[1;32m     31[0m [38;5;66;03m#                    xref="paper", yref="paper",[39;00m
[1;32m     32[0m [38;5;66;03m#                    align='right',[39;00m
[1;32m     33[0m [38;5;66;03m#                    showarrow=False,[39;00m
[1;32m     34[0m [38;5;66;03m#                    xanchor='right', yanchor='bottom',[39;00m
[1;32m     35[0m [38;5;66;03m#                    text='some fancy caption here')[39;00m
[1;32m     37[0m fig[38;5;241m.[39mshow()

[0;31mNameError[0m: name 'series' is not defined
NameError: name 'series' is not defined

